src/api/app/components/token_card_component.html.haml

```diff
@@ -1,37 +1,43 @@
-.card.m-1.p-2
-  .card-body
+= render(CardComponent.new) do |component|
+  %p.card-text
+    Id: #{@token.id}
+  %p.card-text
+    = operation
+  - if @token.package
+    %p.card-text.text-word-break-all
+      Package:
+      = token_package_link
+  %p.card-text
+    Runs as: #{link_to(@token.executor.login, user_path(@token.executor))}
+  - if @token.triggered_at.present?
+    %p.card-text
+      Last trigger on #{@token.triggered_at}
+
+  - component.with_header do
     = link_to token_path(@token) do
       - if @token.description.present?
         %h5.card-title
           = @token.description
       - else
         %h5.card-title.font-italic No description
-    %p.card-text
-      Id: #{@token.id}
-    %p.card-text
-      = operation
-    - if @token.package
-      %p.card-text.text-word-break-all
-        Package:
-        = token_package_link
-    %p.card-text
-      Runs as: #{link_to(@token.executor.login, user_path(@token.executor))}
-    - if @token.triggered_at.present?
-      %p.card-text
-        Last trigger on #{@token.triggered_at}
-    .row
-      .col
-        - if policy(@token).edit?
-          = link_to(edit_token_path(@token), title: 'Edit Token') do
-            %i.fas.fa-edit
-        - if policy(@token).webui_trigger?
-          = link_to(token_trigger_path(@token), title: 'Trigger Token') do
-            %i.fas.fa-project-diagram
-        = link_to '#', title: 'Delete Token',
-          data: { 'bs-toggle': 'modal', 'bs-target': '#delete-token-modal', token_id: @token.id, action: token_path(@token) } do
-          %i.fas.fa-times-circle.text-danger
-        - if @token.type == 'Token::Workflow'
-          = link_to(token_users_path(@token.id), title: 'Share Token') do
-            %i.fas.fa-users
-          = link_to(token_workflow_runs_path(@token.id), title: 'Workflow Runs') do
-            %i.fas.fa-book-open
+
+  - component.with_delete_button do
+    = link_to '#', title: 'Delete Token',
+      data: { 'bs-toggle': 'modal', 'bs-target': '#delete-token-modal', token_id: @token.id, action: token_path(@token) } do
+      %i.fas.fa-times-circle.text-danger
+
+  - component.with_action do
+    - if policy(@token).edit?
+      = link_to(edit_token_path(@token), title: 'Edit Token', class: 'nav-link p-1') do
+        %i.fas.fa-edit
+  - component.with_action do
+    - if policy(@token).webui_trigger?
+      = link_to(token_trigger_path(@token), title: 'Trigger Token', class: 'nav-link p-1') do
+        %i.fas.fa-project-diagram
+  - component.with_action do
+    - if @token.type == 'Token::Workflow'
+      = link_to(token_users_path(@token.id), title: 'Share Token', class: 'nav-link p-1') do
+        %i.fas.fa-users
+  - component.with_action do
+    = link_to(token_workflow_runs_path(@token.id), title: 'Workflow Runs', class: 'nav-link p-1') do
```

Comment:
@ncounter i think this also has to go under a `if @token.type == 'Token::Workflow'`

Reply:
only available for workflow tokens

Reply:
good catch, I probably overlooked it, thank you :+1